js/modulestest/compile.go

```diff
@@ -0,0 +1,43 @@
+package modulestest
+
+import (
+	"io"
+	"os"
+	"path"
+	"path/filepath"
+
+	"github.com/dop251/goja"
+)
+
+// CompileFile compiles a JS file as a [*goja.Program].
+func CompileFile(base, name string) (*goja.Program, error) {
+	filename := path.Join(base, name)
+
+	//nolint:forbidigo // Allow os.Open in tests
+	f, err := os.Open(filepath.Clean(filename))
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		err = f.Close()
+		if err != nil {
+			panic(err)
+		}
+	}()
+
+	b, err := io.ReadAll(f)
+	if err != nil {
+		return nil, err
+	}
+
+	return compile(name, b)
+}
+
+func compile(name string, b []byte) (*goja.Program, error) {
+	program, err := goja.Compile(name, string(b), false)
+	if err != nil {
+		return nil, err
+	}
+
+	return program, nil
+}
```

Comment:
I reused this snippet from what I did already in the past in webcrypto. If I remember, at the time I copied this pattern from the k6 codebase, it might have been linked to tc39 indeed, I can't remember. I think I did that at the time to make evaluating a couple of scripts involved in testing more convenient.

I'll look into it again 👀 and switch to ReadFile 👍🏻

Reply:
I remembered as I modifed it what made me create this helper at the time. The goal was really to fill the gap for the lack of `goja.Runtime.RunFile` or something like that. I think it's reasonably useful in the context of testing extensions like here or webcrypto, although it could probably use a different name to align with the Program terminology.

I've rewritten the function to use `os.ReadFile`, simplify it, and document it 👍🏻